crates/core/src/wasm_runtime/runtime.rs

```diff
@@ -260,7 +266,19 @@ impl Runtime {
 
     fn instance_store() -> Store {
         use wasmer::Cranelift;
-        Store::new(Cranelift::new())
+        use wasmer_middlewares::Metering;
+
+        // Simple metering middleware that counts the number of instructions executed
+        let initial_limit = 10_000;
```

Comment:
10_000 instructions is extremelly low, would be good to have. arought idea on what is a reasoably initial setup, I am almost sure is in the 1000's of millions at the very least. not all insttructions are created equal but let's assume an expensive set and say a contract shouldn't take longer than 20 seconds to return and go from there.

When we optimize the executor this will be less of an issue.